### PR TITLE
feat: [CO-775] Remove Zimbra Mobile Gateway feature

### PR DESCRIPTION
**Pull request by Polpetta**
ref: [CO-775]

### CODE REVIEW

File: common/src/main/java/com/zimbra/common/soap/AccountConstants.java [hunk 1]
```diff
@@ -124,9 +124,6 @@ private AccountConstants() {
   public static final String E_GET_SMIME_PUBLIC_CERTS_REQUEST = "GetSMIMEPublicCertsRequest";
   public static final String E_GET_SMIME_PUBLIC_CERTS_RESPONSE = "GetSMIMEPublicCertsResponse";
 
-  // Zimbra Mobile Gateway
-  public static final String E_BOOTSTRAP_MOBILE_GATEWAY_APP_REQUEST =
-      "BootstrapMobileGatewayAppRequest";
   public static final String E_BOOTSTRAP_MOBILE_GATEWAY_APP_RESPONSE =
```
Comment (frisonisland) [inline common/src/main/java/com/zimbra/common/soap/AccountConstants.java:127]: I think we are missing these

File: store/src/main/java/com/zimbra/cs/service/mail/CreateDataSource.java [hunk 2]
```diff
@@ -26,251 +17,257 @@
...
+  public Element handle(Element request, Map<String, Object> context) throws ServiceException {
+    ZimbraSoapContext zsc = getZimbraSoapContext(context);
+    Provisioning prov = Provisioning.getInstance();
 
-        Element eDataSource = getDataSourceElement(request);
-        DataSourceType type = DataSourceType.fromString(eDataSource.getName());
+    Element eDataSource = getDataSourceElement(request);
+    DataSourceType type = DataSourceType.fromString(eDataSource.getName());
 
-        boolean isAppProvisioned = doZMGAppProvisioningIfReq(zsc, prov, eDataSource, type);
```
Comment (frisonisland) [inline store/src/main/java/com/zimbra/cs/service/mail/CreateDataSource.java:45]: I think you can remove this, as the only condition where it is used evaluates always to true

File: store/utils/docker/carbonio-db/scripts/01-db.sql [hunk 3]
```diff
@@ -224,23 +224,6 @@ CREATE TABLE current_sessions (
 	PRIMARY KEY (id, server_id)
 ) ENGINE = InnoDB;
 
--- ZMG Devices
-CREATE TABLE zmg_devices (
```
[the remark this answers is not in the document]
Comment (Polpetta) [reply store/utils/docker/carbonio-db/scripts/01-db.sql:228]: Ok, I'll drop it there too :+1:

File: store/src/main/java/com/zimbra/cs/util/GalSyncAccountUtil.java [hunk 4]
```diff
@@ -30,312 +19,360 @@
...
+    } catch (HttpException e) {
 
-	private static Map<String,Integer> mCommands;
+    } finally {
+      if (mTransport != null) mTransport.shutdown();
+    }
+  }
 
-	private static void usage() {
-		System.out.println("zmgsautil: {command}");
```
[the remark this answers is not in the document]
Comment (Polpetta) [reply store/src/main/java/com/zimbra/cs/util/GalSyncAccountUtil.java:52]: Yeah, my goal was to remove the first line, I think I got carried away by the deletion process hehe

**Review by frisonisland — CHANGES_REQUESTED** [review]
There are still some references that can be removed, everything else looks good to me

File: common/src/main/java/com/zimbra/common/soap/AccountConstants.java (hunk 1 above)
Comment (Polpetta) [reply common/src/main/java/com/zimbra/common/soap/AccountConstants.java]: Removed!

Comment (frisonisland) [reply common/src/main/java/com/zimbra/common/soap/AccountConstants.java]: All of them :laughing: See all "MOBILE_GATEWAY" constants!

Comment (Polpetta) [reply common/src/main/java/com/zimbra/common/soap/AccountConstants.java]: Done

**Review by keshavbhatt — APPROVED** [review]
LGTM.

**Review by aheeva-yuliya — APPROVED** [review]

**Review by frisonisland — APPROVED** [review]

**Pull request merged**